go-controller/pkg/ovn/egressip_test.go

```diff
@@ -612,12 +612,14 @@ var _ = ginkgo.Describe("OVN master EgressIP Operations", func() {
 				_, err = fakeOvn.fakeClient.KubeClient.CoreV1().Nodes().Update(context.TODO(), &node2, metav1.UpdateOptions{})
 				gomega.Expect(err).NotTo(gomega.HaveOccurred())
 				// sleep long enough for TransactWithRetry to fail, causing egressnode operations to fail
-				time.Sleep(types.OVSDBTimeout + time.Second)
+				// there is a chance that both egressnode events(node1 removal and node2 update) will end up in the same event queue
+				// sleep for double the time to allow for two consecutive TransactWithRetry timeouts
```

Comment:
so there is one question I have been meaning to ask, these event queues are per resource right? So `nodes` will have 15 queues which are separate from the ones for pods and namespaces right? If yes, then how can a 3 node cluster end up having 2 nodes landings on the same queue...oh maybe because the node updates happen very often from kubelet so its possible the other queues are already filled... hmm

Reply:
This is the code for choosing a queue number for a new object:
```
func (i *informer) getNewQueueNum(numEventQueues uint32) uint32 {
	var j, startIdx, queueIdx uint32
	startIdx = uint32(rand.Intn(int(numEventQueues - 1)))
	queueIdx = startIdx
	lowestNum := len(i.events[startIdx])
	for j = 0; j < numEventQueues; j++ {
		tryQueue := (startIdx + j) % numEventQueues
		num := len(i.events[tryQueue])
		if num < lowestNum {
			lowestNum = num
			queueIdx = tryQueue
		}
	}
	return queueIdx
}
```
So it chooses a random queue with the least assigned events.
There is a small chance that the same queue will be used for both events and I think this is what we are seeing here.

Reply:
All of the following took 20.x seconds:
https://github.com/ovn-org/ovn-kubernetes/runs/6885568095?check_suite_focus=true#step:6:24691
https://github.com/ovn-org/ovn-kubernetes/runs/7032634790?check_suite_focus=true#step:6:9418
https://github.com/ovn-org/ovn-kubernetes/runs/6425275234?check_suite_focus=true#step:6:21353

Logs also confirm that the events were handled in sequence